iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/helplets.inl

```diff
@@ -155,8 +157,8 @@ inline bool doesEndWithPathSeparator(const string<StringCapacity>& name) noexcep
     {
         return false;
     }
-
-    char lastCharacter = name[name.size() - 1U];
+    // AXIVION Next Construct AutosarC++19_03-A3.9.1: Not used as an integer but as actual character
```

Comment:
@saif-at-github @FerdinandSpitzschnueffler 

Why do we need to justify anything here? We extract a char from a char array, that should be very valid and safe code. Either I miss something important, or it is kind of a problem to have this rule if it prohibits using chars the way they are intended to.

Maybe there is a good explanation, but I cannot provide it.

Reply:
@MatthiasKillat This code `char lastCharacter = name[name.size() - 1U];` raise the violation `The basic numerical types` are not supposed to be used , instead use specific-length types. This rule is prohibits to using char type.

Reply:
@saif-at-github I understand that, I just do not think we should keep the rule then and suppress it as it is a natural use case for `char`. Littering everything with comments because of a bad rule is just not good style.

OK for now, as it is not decided what to d with this rule.